internal/helm/helm.go

```diff
@@ -197,6 +203,55 @@ func newChartFromDirectory(directory string) (*chart.Chart, error) {
 	return loader.LoadDir(directory)
 }
 
+// newChartFromRemoteWithCache fetches the chart from remote repository, the chart file
+// is then stored in the local cache directory for future usage.
+func newChartFromRemoteWithCache(ciliumVersion semver2.Version) (*chart.Chart, error) {
+	cacheDir, err := ciliumCacheDir()
+	if err != nil {
+		return nil, err
+	}
+
+	file := path.Join(cacheDir, fmt.Sprintf("cilium-%s.tgz", ciliumVersion))
+	if _, err = os.Stat(file); err != nil {
```

Comment:
nit: I would directly try `os.Open` here and fallback to fetching on `fs.ErrNotExist`. I don't see an problem here, but this kind of pattern can run into TOCTOU issues.

Reply:
I park this one right now, as we might eventually need to read the file one more time.